src/lib.rs

```diff
@@ -1715,8 +1920,34 @@ pub unsafe trait FromBytes: FromZeroes {
 
     /// Interprets the given `bytes` as a `&mut Self` without copying.
     ///
-    /// If `bytes.len() != size_of::<T>()` or `bytes` is not aligned to
-    /// `align_of::<T>()`, this returns `None`.
+    /// If `bytes.len() != size_of::<Self>()` or `bytes` is not aligned to
+    /// `align_of::<Self>()`, this returns `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use zerocopy::FromBytes;
+    /// # use zerocopy_derive::*;
+    ///
+    /// #[derive(AsBytes, FromZeroes, FromBytes)]
+    /// #[repr(C)]
+    /// struct PacketHeader {
+    ///     src_port: [u8; 2],
+    ///     dst_port: [u8; 2],
+    ///     length: [u8; 2],
+    ///     checksum: [u8; 2],
+    /// }
+    ///
+    /// // These bytes encode a `PacketHeader`.
+    /// let bytes = &mut [0, 1, 2, 3, 4, 5, 6, 7][..];
+    ///
+    /// let header = PacketHeader::mut_from(bytes).unwrap();
+    ///
+    /// assert_eq!(header.src_port, [0, 1]);
+    /// assert_eq!(header.dst_port, [2, 3]);
+    /// assert_eq!(header.length, [4, 5]);
+    /// assert_eq!(header.checksum, [6, 7]);
```

Comment:
Maybe also perform mutation to demonstrate that the memory is shared? Something like:

```rust
header.checksum = [0, 0];

assert_eq!(bytes, &[0, 1, 2, 3, 4, 5, 0, 0]);
```

Same below with other `mut` methods.

Reply:
Done!